### PR TITLE
Reuse Grid.__init__ in ImageGrid.__init__.

### PR DESCRIPTION
**Pull request by anntzer**
`ImageGrid.__init__` differs from `Grid.__init__` by some defaults which
can be passed to the superclass, and by additionally constructing the
colorbar axes.  This needs to be pushed into `_update_locators` which
isn't optimal but still better than duplicating the entire
`Grid.__init__`.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by timhoffm** [comment]
Initializing the colorbar in `_update_locators()` is unexpected.

Can you make a separate method (something like `_update_extra_ax`, name up for improvement) for updating the colorbar. The method is empty in `Grid`. IMHO this is better than doing unrelated stuff in `_update_locators`.

Alternatively, you could find a more general name instead of `_update_locators`.

BTW: Is it `_update_locators` or rather `_init_locators()`? Currently this is only called during setup; and I think it's not feasible to use this method anywhere again later (currently it is not).

**Comment by anntzer** [comment]
I just renamed it to _init_locators, as that's what it really is.

File: lib/mpl_toolkits/axes_grid1/axes_grid.py [hunk 1]
```diff
@@ -485,9 +434,18 @@ def __init__(self, fig,
                 for ax, cax in zip(self.axes_all, self.cbar_axes):
                     ax.cax = cax
 
-        self.set_label_mode(label_mode)
-
-    def _update_locators(self):
+    def _init_locators(self):
+        if self._colorbar_pad is None:
```
Comment (timhoffm) [inline lib/mpl_toolkits/axes_grid1/axes_grid.py]: Add a comment along the lines:

> Slightly abusing this method as a way to inject colorbar creation into the intialization.

Comment (anntzer) [reply lib/mpl_toolkits/axes_grid1/axes_grid.py]: sure

**Pull request merged**